ui/apps/platform/src/Components/CompoundSearchFilter/components/CompoundSearchFilterInputField.tsx

```diff
@@ -121,13 +154,18 @@ function CompoundSearchFilterInputField({
         const attributeLabel = attributeObject.displayName.toLowerCase();
         const selection = ensureStringArray(value);
         const selectOptions = attributeObject.inputProps.options;
+        const { searchTerm } = attributeObject;
 
         return (
             <CheckboxSelect
-                selection={selection}
-                onChange={(value) => {
+                selection={searchFilter[searchTerm]}
+                onChange={(checked, _value) => {
                     onChange(value);
-                    onSearch(value);
+                    onSearch({
+                        action: checked ? 'ADD' : 'REMOVE',
+                        category: attributeObject.searchTerm,
+                        value: _value,
+                    });
```

Comment:
I think the `isSelected` prop for the individual options needs to be modified to use the `searchFilter` as well.

Reply:
Also we should have a `key` prop on the options.

Reply:
True. We don't need that old `selection` variable anymore. I'll take the value returned from `searchFilter[searchTerm]` ensure it's a string array and then use that instead